src/components/Calculator.js

```diff
@@ -17,9 +17,32 @@ class Calculator extends Component {
     this.setState({ a: '0', b: '0', operation: '', displayed: '0' });
   }
   
+  addDigit(digit) {
+    const { a, b, operation } = this.state;
+    if (operation === '') {
+      if (a === '0') {
+        this.setState({ a: digit, displayed: digit });
+      } else if (a.length < 9) {
+        this.setState({ a: a.concat(digit), displayed: a.concat(digit) });
+      }
+    } else {
+      if (b === 0) {
```

Comment:
You initialised `b` with a string, but here you check if it is a number. You should be consistent with the data type.